kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraWindowedTable.java

```diff
@@ -370,59 +561,67 @@ public long approximateNumEntries(final int partition) {
   @Override
   @CheckReturnValue
   public BoundStatement insert(
-      final int partitionKey,
+      final int kafkaPartition,
       final Stamped<Bytes> key,
       final byte[] value,
       final long epochMillis
   ) {
+    pendingFlushStreamTime = Math.max(pendingFlushStreamTime, key.stamp);
+    final SegmentPartition remotePartition = partitioner.tablePartition(kafkaPartition, key);
     return insert
         .bind()
-        .setInt(PARTITION_KEY.bind(), partitionKey)
+        .setInt(PARTITION_KEY.bind(), remotePartition.partitionKey)
+        .setLong(SEGMENT_ID.bind(), remotePartition.segmentId)
         .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.key.get()))
         .setInstant(WINDOW_START.bind(), Instant.ofEpochMilli(key.stamp))
         .setByteBuffer(DATA_VALUE.bind(), ByteBuffer.wrap(value));
   }
 
   /**
-   * @param partitionKey  the partitioning key
-   * @param key           the data key
+   * @param kafkaPartition  the kafka partition
+   * @param key             the data key
    *
    * @return a statement that, when executed, will delete the row
-   *         matching {@code partitionKey} and {@code key} in the
+   *         matching {@code kafkaPartition} and {@code key} in the
    *         {@code table}. Note that the {@code key} here is the
    *         "windowed key" which includes both the record key and
    *         also the window start timestamp
    */
   @Override
   @CheckReturnValue
   public BoundStatement delete(
-      final int partitionKey,
+      final int kafkaPartition,
       final Stamped<Bytes> key
   ) {
+    final SegmentPartition segmentPartition = partitioner.tablePartition(kafkaPartition, key);
```

Comment:
do we need to set `pendingFlushStreamTime` here?

Reply:
yep, thanks